mne/io/eeglab/_eeglab.py

```diff
@@ -71,13 +71,96 @@ def _check_for_scipy_mat_struct(data):  # taken from pymatreader.utils
     return data
 
 
-def _readmat(fname, uint16_codec=None):
+def _readmat(fname, uint16_codec=None, preload=False):
     try:
         read_mat = _import_pymatreader_funcs("EEGLAB I/O")
     except RuntimeError:  # pymatreader not installed
-        eeg = loadmat(
-            fname, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
-        )
+        if preload:
+            eeg = loadmat(
+                fname, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
+            )
+        else:
+            # the files in eeglab are always the same field names
+            # the the fields were taken from the eeglab sample reference
+            # available at the eeglab github:
+            # https://github.com/sccn/eeglab/blob/develop/sample_data/eeglab_data.set
+            # The sample reference is the big reference for the field names
+            # in eeglab files, and what is used in the eeglab tests.
+            info_fields = [
+                "setname",
+                "filename",
+                "filepath",
+                "subject",
+                "group",
+                "condition",
+                "session",
+                "comments",
+                "nbchan",
+                "trials",
+                "pnts",
+                "srate",
+                "xmin",
+                "xmax",
+                "times",
+                "icaact",
+                "icawinv",
+                "icasphere",
+                "icaweights",
+                "icachansind",
+                "chanlocs",
+                "urchanlocs",
+                "chaninfo",
+                "ref",
+                "event",
+                "urevent",
+                "eventdescription",
+                "epoch",
+                "epochdescription",
+                "reject",
+                "stats",
+                "specdata",
+                "specicaact",
+                "splinefile",
+                "icasplinefile",
+                "dipfit",
+                "history",
+                "saved",
+                "etc",
```

Comment:
For compactness, maybe a `info_fields = """a b c d ...""".split()` would be cleaner?

Reply:
sure